Improve S2 ESA+Sinergise logic separation

jeremyh opened the pull request:
Fixes issue @dunkgray reported (duplicate image bands being found, expecting inner tiles to be specified)

It now normalises the input paths so that we can be more specific when looking for files inside (rather than open-ended globs that were finding too many).

I've also reduced the huge "if" block, and combined logic from the two variants where possible. Deduplicating this logic showed that we weren't extracting the platform correctly in Sinergise datasets, so its test is updated. Lots of code simplification too.

jeremyh committed:
Simplify S2 path handling

jeremyh committed:
Improve S2 ESA+Sinergise logic separation

Normalise the input paths so that we can be more specific when looking
for inner files (rather than open-ended globs that were finding the
wrong files).

Fixes issue Duncan reported (duplicate image bands found, not finding
inner tiles separately)

Deduplicating the logic also showed that we weren't extracting the
platform correctly in Sinergise datasets. Updated its test.

dunkgray commented:
Excellent, thanks Jeremy.

Now, let's give it a spin.

________________________________
From: Jeremy Hooke ***@***.***>
Sent: Tuesday, 4 May 2021 5:07 PM
To: GeoscienceAustralia/eo-datasets ***@***.***>
Cc: Gray Duncan ***@***.***>; Mention ***@***.***>
Subject: [GeoscienceAustralia/eo-datasets] Improve S2 ESA+Sinergise logic separation (#156)


Fixes issue @dunkgray<https://github.com/dunkgray> reported (duplicate image bands being found, expecting inner tiles to be specified)

It now normalises the input paths so that we can be more specific when looking for files inside (rather than open-ended globs that were finding too many).

I've also reduced the huge "if" block, and combined logic from the two variants where possible. Deduplicating this logic showed that we weren't extracting the platform correctly in Sinergise datasets, so its test is updated. Lots of code simplification too.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/GeoscienceAustralia/eo-datasets/pull/156

Commit Summary

  *   Simplify S2 path handling
  *   Improve S2 ESA+Sinergise logic separation

File Changes

  *   M eodatasets3/prepare/sentinel_l1c_prepare.py<https://github.com/GeoscienceAustralia/eo-datasets/pull/156/files#diff-f2654a7c8eb9ec541004b01e7b12b1a2f0f6db2a18c86c5bed009a61fae37784> (269)
  *   M tests/common.py<https://github.com/GeoscienceAustralia/eo-datasets/pull/156/files#diff-53eb316ef7b18ba47acdd955cdaa7ca5467cd5ba6f5cf8ed92c68a421902d120> (3)
  *   M tests/integration/prepare/test_prepare_esa_sentinel_l1.py<https://github.com/GeoscienceAustralia/eo-datasets/pull/156/files#diff-4e3199a6872ae04e5ca639648f5823f90658acbf6e8e5c1519f863417125077e> (29)
  *   M tests/integration/prepare/test_prepare_sinergise_sentinel_l1.py<https://github.com/GeoscienceAustralia/eo-datasets/pull/156/files#diff-e909b2f11d6b2d0c89fde12714d45a3ee579a7ed01744113d83988bc6077344d> (36)

Patch Links:

  *   https://github.com/GeoscienceAustralia/eo-datasets/pull/156.patch
  *   https://github.com/GeoscienceAustralia/eo-datasets/pull/156.diff

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/GeoscienceAustralia/eo-datasets/pull/156>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAIAHKJRYTZQZHJKT3WPPILTL6ML5ANCNFSM44CEPXBQ>.

Geoscience Australia Disclaimer: This e-mail (and files transmitted with it) is intended only for the person or entity to which it is addressed. If you are not the intended recipient, then you have received this e-mail by mistake and any use, dissemination, forwarding, printing or copying of this e-mail and its file attachments is prohibited. The security of emails transmitted cannot be guaranteed; by forwarding or replying to this email, you acknowledge and accept these risks.

jeremyh committed:
Combine tile metadata parsing, fix sun fields

The 'metadata.xml' and 'mtl_tl.xml' files are the same
metadata format, but we had separate functions that
extracted fields differently between them.

The azimuth/angle were the wrong way around in one!

jeremyh committed:
Use human-readable metadata names

Base them on the inner Xml namespace names
rather than obtuse filenames.

jeremyh committed:
S2 ESA: Use tile processing time, not product time

jeremyh committed:
Read datetime only once in Sinergise

The ESA document is more precise.

jeremyh committed:
Include sinergise fields in known properties

They're not ideal. But we probably don't want to invent
inside the "sinergise:" namespace when we're not Sinergise.

jeremyh committed:
Consistent S2 product name between ESA, Sinergise

jeremyh committed:
Use same logging on L1 and S2 prepare

jeremyh commented:
I noticed a bit more duplication: the sinergise tile `metadata.xml` and esa `MTD_TL.xml` are the same metadata format. So I've combined their function .... and the two were doing things quite differently. One of them had the solar position fields reversed.

Some other properties are now common between the two (but were previously different): tile_id, product_name.

jeremyh committed:
Split out s2 platform logic

jeremyh committed:
Use stable ids for S2 L1

jeremyh commented:
Made generated UUIDs stable too

jeremyh commented:
Strangely @ohehir doesn't seem to be in any of our Github orgs, so can't be added for review.

ohehir commented:
> Strangely @ohehir doesn't seem to be in any of our Github orgs, so can't be added for review.

Who do I talk too to fix this? Plus get write access to tesp repo?

dunkgray commented on eodatasets3/prepare/sentinel_l1c_prepare.py:
```diff
@@ -278,32 +319,61 @@ def prepare_and_write(
+    logging.basicConfig(
+        format="%(asctime)s %(levelname)s %(message)s", level=logging.INFO
+    )
+    for input_path in datasets:
+        in_out_paths = [
+            *(
+                (
+                    "sinergise.com",
+                    # Input is the folder
+                    p.parent,
+                    # Output is an inner metadata file, with the same name as the folder (usually S2A....).
+                    (p.parent / f"{p.parent.stem}.odc-metadata.yaml"),
+                )
+                for p in _rglob_with_self(input_path, "tileInfo.json")
+            ),
+            *(
+                (
+                    "esa.int",
+                    # Input is zip file
+                    p,
+                    # Output is a sibling file with metadata suffix.
+                    p.with_suffix(".odc-metadata.yaml"),
+                )
+                for p in _rglob_with_self(input_path, "*.zip")
+            ),
+        ]
```
Can this part be turned into a function and have some unit tests?  The tests could also be useful to show examples of where the meta-data.yaml should be located to work.

jeremyh replied:
It could! my gut feeling is that the bang-for-buck is too low, as the important logic is exercised by the integration tests already ("call it with input filename, expect output filename", etc), and unit test code here would be larger than the function but not give us much extra useful coverage. 

(I'm sometimes hesitant with unit tests as they tend to be higher maintence than integration tests -- they depend too much on internal implementation structure, so are broken easily. There's quite a few in eodatasets, but they're usually used when low cost to write, high reward in what they cover, or test something extremely unlikely to change)

dunkgray commented:
Yo Darren,
Good to hear from you!  I'm slack at responding sorry.

lockdown hasn't been too back the second time round.  I did some work at Services Australia and I was going into the office for that, so I didn't get too house bound.

 How about we do I lunch catch-up in a fortnight?
13 November, @dfat

Cheers
Duncan

________________________________
From: Darren Hansen ***@***.***>
Sent: Thursday, 16 September 2021 3:44 PM
To: zz [External] ***@***.*** ***@***.***>; Duncan Gray ***@***.***>
Subject: How's the Locky-D going? [SEC=UNOFFICIAL]




Matt and Duncan



Hi there – how are you both going in this seemingly endless Lockdown?



Have been working at home since it first started and all the usual distractions, IT issues and daily virtual meetings that go along with it. But, hard to complain when so many others have lost a lot of their income or jobs.  Hope you’ve successfully navigated all the exposure sites – mostly supermarkets and eateries these days.  Handy to have the University of Canberra close by as there are heaps of sports grounds to walk around/exercise that hardly anyone goes to.  Uni of Canberra and ANU campuses are huge so you can usually find somewhere to walk around with few if any people.  There’s always heaps of people walking and jogging around Bruce so not so good if you want to minimise contact.



Waka and I are all vaccinated having had my second pfzier shot 2 wks ago out at the Canberra airport vaccination hub.  Kai and Joy will get their first shot later this month having just become eligible. They get to go out to the AIS vaccination hub that opened the other week and is just a 10 minute walk from home.  Have you managed to get your shots sorted?



I attended my first virtual auction last weekend as a place in our same street – 2 doors down was on the market, exactly the same make as our place. I knew I’d just be a spectator given the stupid prices people are paying but was still surprised that the place went for $990,000. A 3-bedroom two story townhouse! The people who sold it would be pleased having bought it for $428,000 in 2008.  With Kai turning 22 yrs next month and Joy at 19 yrs – probably better for us to wait until we are empty nesters before buying – 2-bedroom places are much cheaper.



Hopefully, we will be out of lockdown mid-October – would be good to catch up for lunch or coffee again then. Looking forward to warmer days.



Best,



Darren





















Geoscience Australia Disclaimer: This e-mail (and files transmitted with it) is intended only for the person or entity to which it is addressed. If you are not the intended recipient, then you have received this e-mail by mistake and any use, dissemination, forwarding, printing or copying of this e-mail and its file attachments is prohibited. The security of emails transmitted cannot be guaranteed; by forwarding or replying to this email, you acknowledge and accept these risks.

dunkgray commented:
Duncan

Hi there and thanks for the mail - good to hear from you.  Lunch on 12 November (Fri) here at DFAT sounds great.  How about 12:30pm.

Best,

Darren

From: Duncan Gray ***@***.***>
Sent: Friday, 29 October 2021 4:35 PM
To: Darren Hansen ***@***.***>; zz [External] ***@***.*** ***@***.***>; GeoscienceAustralia/eo-datasets ***@***.***>
Cc: Ttt ***@***.***) ***@***.***>
Subject: [EXTERNAL] Re: How's the Locky-D going? [SEC=UNOFFICIAL]


CAUTION: This email originated from outside the organisation. Do not click links or open attachments unless you recognise the sender.
Yo Darren,
Good to hear from you!  I'm slack at responding sorry.

lockdown hasn't been too back the second time round.  I did some work at Services Australia and I was going into the office for that, so I didn't get too house bound.

 How about we do I lunch catch-up in a fortnight?
13 November, @dfat

Cheers
Duncan

________________________________
From: Darren Hansen ***@***.******@***.***>>
Sent: Thursday, 16 September 2021 3:44 PM
To: zz [External] ***@***.******@***.***> ***@***.******@***.***>>; Duncan Gray ***@***.******@***.***>>
Subject: How's the Locky-D going? [SEC=UNOFFICIAL]




Matt and Duncan



Hi there - how are you both going in this seemingly endless Lockdown?



Have been working at home since it first started and all the usual distractions, IT issues and daily virtual meetings that go along with it. But, hard to complain when so many others have lost a lot of their income or jobs.  Hope you've successfully navigated all the exposure sites - mostly supermarkets and eateries these days.  Handy to have the University of Canberra close by as there are heaps of sports grounds to walk around/exercise that hardly anyone goes to.  Uni of Canberra and ANU campuses are huge so you can usually find somewhere to walk around with few if any people.  There's always heaps of people walking and jogging around Bruce so not so good if you want to minimise contact.



Waka and I are all vaccinated having had my second pfzier shot 2 wks ago out at the Canberra airport vaccination hub.  Kai and Joy will get their first shot later this month having just become eligible. They get to go out to the AIS vaccination hub that opened the other week and is just a 10 minute walk from home.  Have you managed to get your shots sorted?



I attended my first virtual auction last weekend as a place in our same street - 2 doors down was on the market, exactly the same make as our place. I knew I'd just be a spectator given the stupid prices people are paying but was still surprised that the place went for $990,000. A 3-bedroom two story townhouse! The people who sold it would be pleased having bought it for $428,000 in 2008.  With Kai turning 22 yrs next month and Joy at 19 yrs - probably better for us to wait until we are empty nesters before buying - 2-bedroom places are much cheaper.



Hopefully, we will be out of lockdown mid-October - would be good to catch up for lunch or coffee again then. Looking forward to warmer days.



Best,



Darren





















Geoscience Australia Disclaimer: This e-mail (and files transmitted with it) is intended only for the person or entity to which it is addressed. If you are not the intended recipient, then you have received this e-mail by mistake and any use, dissemination, forwarding, printing or copying of this e-mail and its file attachments is prohibited. The security of emails transmitted cannot be guaranteed; by forwarding or replying to this email, you acknowledge and accept these risks.
